docs/examples/globaleventshandlerprovider/linkHandlersCalloutUpsell.js

```diff
@@ -1,34 +1,40 @@
 // @flow strict
-import { type Node, useState } from 'react';
+import { useMemo, useCallback, type Node, useState } from 'react';
 import {
   ActivationCard,
   Callout,
   Divider,
   Flex,
   Icon,
-  OnLinkNavigationProvider,
+  GlobalEventsHandlerProvider,
   RadioGroup,
   Upsell,
 } from 'gestalt';
 
 export default function Example(): Node {
   const [onNavigationMode, setOnNavigationMode] = useState<'default' | 'custom'>('default');
 
-  const useOnNavigation = ({ href }: {| href: string, target?: null | 'self' | 'blank' |}) => {
-    const onNavigationClick = ({ event }: {| +event: SyntheticEvent<> |}) => {
-      event.preventDefault();
-      // eslint-disable-next-line no-alert
-      alert(`Disabled link: ${href}. Opening help.pinterest.com instead.`);
-    };
+  const useOnNavigation = useCallback(
+    ({ href }: {| href: string, target?: null | 'self' | 'blank' |}) => {
```

Comment:
Can/should these types reference the type from the relevant component instead of being defined literally?